src/disk_io_thread.cpp

```diff
@@ -941,12 +946,19 @@ namespace libtorrent
 			--pe->piece_refcount;
 		}
 
+		f.lock();
+
+		if (num == 0) return;
```

Comment:
num is a local variable here, presumably this check wouldn't have to be done while holding the mutex. but I guess we need to re-acquire the mutex before returning